include/envoy/tracing/http_tracer.h

```diff
@@ -191,6 +191,7 @@ class HttpTracer {
 };
 
 using HttpTracerPtr = std::unique_ptr<HttpTracer>;
+using HttpTracerSharedPtr = std::shared_ptr<HttpTracer>;
```

Comment:
If we need to work with shared_ptr now for this, do we still need the unique_ptr variant?

Reply:
unique_ptr  is still in use by `TracerFactory` interface and implementations.

should I refactor them to use `std::unique_ptr<HttpTracer>` directly ?

Reply:
I would probably just have the factories return a shared_ptr directly since that is what is ultimately required, but it's not a huge deal.

Reply:
From the design perspective, I would prefer factories to return `unique_ptr` to make it clear to the users of that API that the returned object is always new and the calling code is the only owner for it.

Reply:
That's fine.

Reply:
I've left comments to clarify what is the expected usage for `HttpTracerPtr` and `HttpTracerSharedPtr`